github/commitstatus.go

```diff
@@ -244,15 +254,23 @@ func checkForRetry(res httpResponse, waitTime, maxSleepTime time.Duration) time.
 	// If you exceed the rate limit, the response will have a 403 status and the x-ratelimit-remaining header will be 0
 	// https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#exceeding-the-rate-limit
 	case res.statusCode == http.StatusForbidden && res.rateLimitRemaining == 0:
-		sleepTime := time.Until(res.rateLimitReset)
+		sleepTime := res.rateLimitReset.Sub(res.date) + jitter
```

Comment:
Here we are conflating two different operations in one line and we are missing a comment explaining what is happening. We could do instead (untested):

```go
// Calculate sleeptime based solely on the server clock. This is unaffected by the clock drift between server and client.
sleepTime := res.rateLimitReset.Sub(res.date)
// Be a good netizen by adding some jitter to the time we sleep.
sleepTime := res.rateLimitReset.Sub(res.date) + jitter
```

Reply:
Changed as suggested.